lib/test_cases/function/pull_request_file.js

```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
+                      test: {
+                          category: ['functionality'],
+                          title: 'Add name tags for subscriber services.',
+                          description: 'Services that are special, like event subscribers, route subscriber or access checker, should be tagged with name in services.yml.'
+                      }
+                  }
+                );
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Check for long functions.
+     */
+    function pullRequestFileCheckLongFunctions(prFile) {
+        let result = [];
+        if (prFile.type === 'php' || prFile.type === 'module' || prFile.type === 'theme') {
+            let targetPrLine = undefined;
+            let isFunction = 0;
+            let countFunctionLines = 0;
+            let fnStartLine, fnEndLine = 0;
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("^function", "gm");
+                    let content = currentPrLine.content;
+                    if (pattern.test(content)) {
+                        isFunction = 1;
+                        fnStartLine = currentPrLine.number;
+                        targetPrLine = currentPrLine;
+                    }
+                    if (isFunction) {
+                        if (content.includes("{")) {
+                            countFunctionLines++;
+                        }
+                        if (countFunctionLines > 0 && content.includes("}")) {
+                            countFunctionLines--;
+                        }
+                    }
+                    if (countFunctionLines === 0 && isFunction) {
+                        isFunction = 0;
+                        fnEndLine = currentPrLine.number;
+                        if ((fnEndLine - fnStartLine) > 30) {
+                            result.push(
+                              {
+                                  line: targetPrLine,
+                                  test: {
+                                      category: ['functionality'],
+                                      title: 'Avoid long functions if possible.',
+                                      description: 'Kindly keep the function length to 30 LOC.'
+                                  }
+                              }
+                            );
+                            targetPrLine = undefined;
+                        }
+                    }
+                }
+            })
+        }
+
+        return result;
+    }
+
+    /**
+     * Check for internal URLs.
+     */
+    function pullRequestFileCheckInternalUrls(prFile) {
+        let result = [];
+        if (prFile.type === 'js') {
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("[a-z\\-0-9]+\\/");
```

Comment:
Simply relying on this won't be sufficient, we can check if the life also has http, href or some relevant checks.

Reply:
link, a tag ...or something similar

Reply:
url